pkg/services/ngalert/api/api_ruler_validation.go

```diff
@@ -177,6 +177,7 @@ func validateRuleGroup(
 			}
 			uids[rule.UID] = idx
 		}
+		rule.RuleGroupIndex = idx + 1
```

Comment:
Why do we modify the rule as part of rule validation?

Reply:
we do not modify it. The domain model of the rule is created above.